app/routes/application.js

```diff
@@ -1,17 +1,17 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
 
-export default Route.extend({
-  flashMessages: service(),
-  session: service(),
+export default class ApplicationRoute extends Route {
+  @service flashMessages;
+  @service session;
 
   beforeModel() {
     this.session.loadUser();
-  },
+  }
 
-  actions: {
-    didTransition() {
-      this.flashMessages.step();
-    },
-  },
-});
+  @action
```

Comment:
I'm a little worried about this because I've had issues with the `@action` decorator on routes in the past 🤔

Reply:
Do you remember what the issue was? I tried checking the literature and this seemed the correct way to upgrade :/

Reply:
nope, unfortunately I don't remember the specifics, but I'd sleep better if I knew that we had better test coverage for these things :-/